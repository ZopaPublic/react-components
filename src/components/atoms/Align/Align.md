### Summary

`<Align />` provides textual alignment of elements in a page at different breakpoints.

### Example

- This will align the text to the center at mobile (xs) and right at tablet (m)

```tsx
import { Align } from '@zopauk/react-components';

<Align xs="center" m="right">
  You can save with us if you have tax residency in the UK only. <a href="#">Find out more</a>
</Align>;
```
