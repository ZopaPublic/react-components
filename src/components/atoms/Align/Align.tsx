import React, { HTMLAttributes } from 'react';
import styled from 'styled-components';

import { grid } from '../../../constants';
import { GridBreakpoints } from '../../../constants/grid';

export interface AlignProps extends HTMLAttributes<HTMLDivElement> {
  children?: React.ReactNode;
  xs?: string;
  s?: string;
  m?: string;
  l?: string;
  xl?: string;
}

const Alignment = styled.div<AlignProps>`
  ${(props) =>
    Object.keys(grid.breakpoints)
      .sort((a, b) => grid.breakpoints[a as GridBreakpoints] - grid.breakpoints[b as GridBreakpoints])
      .filter((breakpoint) => Object.keys(props).includes(breakpoint))
      .map(
        (breakpoint) => `
        @media (min-width: ${grid.breakpoints[breakpoint as GridBreakpoints]}px) {
          text-align: ${props[breakpoint as GridBreakpoints]};
        }
        `,
      )}
`;

const Align = (props: AlignProps) => <Alignment {...props} />;

export default Align;
