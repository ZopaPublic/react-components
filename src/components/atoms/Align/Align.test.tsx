import axe from '../../../../axe-helper';
import React from 'react';
import { render } from '@testing-library/react';
import Align from './Align';

describe('Align', () => {
  it('renders the component with no props and no a11y violations', async () => {
    const { container } = render(<Align>Align Content</Align>);
    expect(container.firstChild).toMatchSnapshot();
    const results = await axe(container.innerHTML);
    expect(results).toHaveNoViolations();
  });

  it.each`
    alignment
    ${'left'}
    ${'center'}
    ${'right'}
  `('renders with specific severity: $severity', ({ alignment }) => {
    const { container } = render(<Align xs={alignment}>Align Content</Align>);
    expect(container.firstChild).toMatchSnapshot();
  });
});
