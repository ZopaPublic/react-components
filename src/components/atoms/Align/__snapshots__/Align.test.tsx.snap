// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Align renders the component with no props and no a11y violations 1`] = `
<div
  class=""
>
  Align Content
</div>
`;

exports[`Align renders with specific severity: $severity 1`] = `
@media (min-width:0px) {
  .c0 {
    text-align: left;
  }
}

<div
  class="c0"
>
  Align Content
</div>
`;

exports[`Align renders with specific severity: $severity 2`] = `
@media (min-width:0px) {
  .c0 {
    text-align: center;
  }
}

<div
  class="c0"
>
  Align Content
</div>
`;

exports[`Align renders with specific severity: $severity 3`] = `
@media (min-width:0px) {
  .c0 {
    text-align: right;
  }
}

<div
  class="c0"
>
  Align Content
</div>
`;
